### PR TITLE
samples: openthread: Add nrf54l15 targets to NCS.

### PR DESCRIPTION
**Pull request by canisLupus1313**
This commit adds nrf54l15 support to coprocessor and cli samples.

### CODE REVIEW

File: samples/openthread/coprocessor/boards/nrf54l15pdk_nrf54l15_cpuapp.conf [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#
+# Copyright (c) 2023 Nordic Semiconductor
```
Comment (maciejbaczmanski) [inline samples/openthread/coprocessor/boards/nrf54l15pdk_nrf54l15_cpuapp.conf:2]:
```suggestion
# Copyright (c) 2024 Nordic Semiconductor
```

File: samples/openthread/coprocessor/boards/nrf54l15pdk_nrf54l15_cpuapp.overlay [hunk 2]
```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
```
Comment (maciejbaczmanski) [inline samples/openthread/coprocessor/boards/nrf54l15pdk_nrf54l15_cpuapp.overlay:2]:
```suggestion
 * Copyright (c) 2024 Nordic Semiconductor ASA
```

File: samples/openthread/coprocessor/prj.conf [hunk 3]
```diff
@@ -12,7 +12,7 @@ CONFIG_OPENTHREAD_COPROCESSOR_RCP=y
 CONFIG_LOG_PROCESS_THREAD_STACK_SIZE=2048
 
 # Enable OpenThread features set
-CONFIG_OPENTHREAD_NORDIC_LIBRARY_MASTER=y
+#CONFIG_OPENTHREAD_NORDIC_LIBRARY_MASTER=y
```
Comment (maciejbaczmanski) [inline samples/openthread/coprocessor/prj.conf:15]: Should this be commented out? or removed?

File: samples/openthread/cli/boards/nrf54l15pdk_nrf54l15_cpuapp.conf [hunk 4]
```diff
@@ -0,0 +1,20 @@
+#
+# Copyright (c) 2023 Nordic Semiconductor
```
Comment (maciejbaczmanski) [inline samples/openthread/cli/boards/nrf54l15pdk_nrf54l15_cpuapp.conf:2]:
```suggestion
# Copyright (c) 2024 Nordic Semiconductor
```

File: samples/openthread/coprocessor/boards/nrf54l15pdk_nrf54l15_cpuapp.overlay [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+&uart30 {
+    current-speed = <115200>;
+    status = "okay";
+    hw-flow-control;
+
```
Comment (maciejbaczmanski) [inline samples/openthread/coprocessor/boards/nrf54l15pdk_nrf54l15_cpuapp.overlay:11]:
```suggestion
```

**Review by maciejbaczmanski — COMMENTED** [review]
Thanks! few nits below

File: samples/openthread/coprocessor/boards/nrf54l15pdk_nrf54l15_cpuapp.conf (hunk 1 above)
Comment (canisLupus1313) [reply samples/openthread/coprocessor/boards/nrf54l15pdk_nrf54l15_cpuapp.conf:2]: fixed.

File: samples/openthread/coprocessor/prj.conf (hunk 3 above)
Comment (canisLupus1313) [reply samples/openthread/coprocessor/prj.conf:15]: fixed.

File: samples/openthread/cli/boards/nrf54l15pdk_nrf54l15_cpuapp.conf (hunk 4 above)
Comment (canisLupus1313) [reply samples/openthread/cli/boards/nrf54l15pdk_nrf54l15_cpuapp.conf:2]: fixed.

File: samples/openthread/coprocessor/boards/nrf54l15pdk_nrf54l15_cpuapp.overlay (hunk 5 above)
Comment (canisLupus1313) [reply samples/openthread/coprocessor/boards/nrf54l15pdk_nrf54l15_cpuapp.overlay]: fixed.

**Comment by edmont** [comment]
I believe samples documentation and NCS changelog should be updated. Also, are CoAP samples not contemplated for nRF54L15 or just deferred?

**Review by maciejbaczmanski — APPROVED** [review]

**Comment by canisLupus1313** [comment]
> I believe samples documentation and NCS changelog should be updated. Also, are CoAP samples not contemplated for nRF54L15 or just deferred?

CoAP samples are currently not in the scope of any dev tag so we aren't working on those.  

As for dock and change log You are wright I will update those.

**Review by kszewenko — APPROVED** [review]

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 6]
```diff
@@ -97,7 +97,7 @@ The following list summarizes the most important changes inherited from the upst
 Thread
 ------
 
-|no_changes_yet_note|
+* Initial experimental support for CLI and coprosessor samples for nRF54l15.
```
Comment (greg-fer) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:100]:
```suggestion
* Initial experimental support for nRF54L15 for Thread CLI and Co-processor samples.
```
Same information needs to be propagated in the Sample section for Thread samples, similarly to https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/releases_and_maturity/releases/release-notes-changelog.html#nfc-samples .

**Review by greg-fer — CHANGES_REQUESTED** [review]

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst (hunk 6 above)
Comment (canisLupus1313) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst]: fixed.

**Review by greg-fer — CHANGES_REQUESTED** [review]
Second part of my comment is not addressed.

**Review by greg-fer — APPROVED** [review]

**Review by edmont — APPROVED** [review]

**Pull request merged**